sdk/authorization/Azure.ResourceManager.Authorization/src/Generated/RoleAssignmentData.cs

```diff
@@ -24,13 +25,57 @@ internal RoleAssignmentData()
+        /// <param name="delegatedManagedIdentityResourceId"> Id of the delegated managed identity resource. </param>
+        internal RoleAssignmentData(ResourceIdentifier id, string name, ResourceType resourceType, SystemData systemData, string scope, ResourceIdentifier roleDefinitionId, Guid? principalId, RoleAssignmentPrincipalType? principalType, string description, string condition, string conditionVersion, DateTimeOffset? createdOn, DateTimeOffset? updatedOn, string createdBy, string updatedBy, string delegatedManagedIdentityResourceId) : base(id, name, resourceType, systemData)
         {
-            Properties = properties;
+            Scope = scope;
+            RoleDefinitionId = roleDefinitionId;
+            PrincipalId = principalId;
+            PrincipalType = principalType;
+            Description = description;
+            Condition = condition;
+            ConditionVersion = conditionVersion;
+            CreatedOn = createdOn;
+            UpdatedOn = updatedOn;
+            CreatedBy = createdBy;
+            UpdatedBy = updatedBy;
+            DelegatedManagedIdentityResourceId = delegatedManagedIdentityResourceId;
         }
 
-        /// <summary> Role assignment properties. </summary>
-        public RoleAssignmentPropertiesWithScope Properties { get; }
+        /// <summary> The role assignment scope. </summary>
+        public string Scope { get; }
+        /// <summary> The role definition ID. </summary>
+        public ResourceIdentifier RoleDefinitionId { get; }
+        /// <summary> The principal ID. </summary>
+        public Guid? PrincipalId { get; }
+        /// <summary> The principal type of the assigned principal ID. </summary>
+        public RoleAssignmentPrincipalType? PrincipalType { get; }
+        /// <summary> Description of role assignment. </summary>
+        public string Description { get; }
+        /// <summary> The conditions on the role assignment. This limits the resources it can be assigned to. e.g.: @Resource[Microsoft.Storage/storageAccounts/blobServices/containers:ContainerName] StringEqualsIgnoreCase &apos;foo_storage_container&apos;. </summary>
+        public string Condition { get; }
+        /// <summary> Version of the condition. Currently the only accepted value is &apos;2.0&apos;. </summary>
+        public string ConditionVersion { get; }
+        /// <summary> Time it was created. </summary>
+        public DateTimeOffset? CreatedOn { get; }
+        /// <summary> Time it was updated. </summary>
+        public DateTimeOffset? UpdatedOn { get; }
+        /// <summary> Id of the user who created the assignment. </summary>
+        public string CreatedBy { get; }
+        /// <summary> Id of the user who updated the assignment. </summary>
+        public string UpdatedBy { get; }
+        /// <summary> Id of the delegated managed identity resource. </summary>
+        public string DelegatedManagedIdentityResourceId { get; }
```

Comment:
Should it be an `ResourceIdentifier`?

Reply:
Pending on service team's confrm, from test it's more like a raw string